### ISSUE TITLE
[Disk] HD images no longer mounted

### ISSUE THREAD

**Issue by Michaelangel007** [AppleWin/AppleWin#348]
Verified this is a regression from 1.25.
HD images are not mounted and not displayed in the `Disk` configuration dialog even though they are selected.

https://groups.google.com/d/msg/comp.emulators.apple2/rykZ0DyIBkI/Y8mSjLG9EwAJ
```
No more HD emulation with AppleWin1.26.0.2 ?

I can not mount images > 800K. No matter if they are .2mg, .po or .hdv. 
Is something happened or am I doing some stupid mistake ? 

Thank you 
Marco 
```

**Comment by tomcw** [comment]
This was broken by me back in 717c5cb (Feb 2015!) - but only in master (not in any release).

New pre-release here:
https://github.com/AppleWin/AppleWin/releases/tag/v1.26.0.3

**Comment by Michaelangel007** [comment]
That was a fast fix ! Thanks !  I was about to ask what commit broken this too but I see you addressed that too!

Minor quibble:

* We probably should fix the formatting on Lines 62-63

https://github.com/AppleWin/AppleWin/blob/0a7a0e9e5c637f6fbdbfdc7685930ffa09466707/source/DiskImage.cpp#L62-L63

From:

```
	ImageError_e Err = (!bIsHarddisk) ?     sg_DiskImageHelper.Open(pszImageFilename, pImageInfo, bCreateIfNecessary, strFilenameInZip)
									  : sg_HardDiskImageHelper.Open(pszImageFilename, pImageInfo, bCreateIfNecessary, strFilenameInZip);
```

To:

```
	ImageError_e Err = (!bIsHarddisk) 
									  ?     sg_DiskImageHelper.Open(pszImageFilename, pImageInfo, bCreateIfNecessary, strFilenameInZip)
									  : sg_HardDiskImageHelper.Open(pszImageFilename, pImageInfo, 
```

i.e. Horizontally align the `?` and `:` on separate lines.

**Comment by tomcw** [comment]
Hi Michael,
I understand your formatting point. I'll re-open this issue to address this.
btw. I posted to cea2 news-group asking for people to try this build.
Thanks.

**Comment by Michaelangel007** [comment]
Awesome thanks.  Sorry to be a stickler but better to fix this now while it is still fresh then liter up the code with "TODOs" :-)

I was wondering when you were going to announce the 1.26 beta. Thanks for taking care of that too.

**Comment by tomcw** [comment]
Fixed in 6269651 (1.26.0.4).

**Comment by tomcw** [comment]
NB. Added a save-state test to the AppleWin-Test repo here: 5dfdf71
>Added new tests for >800K HDD image...

**Comment by tomcw** [comment]
Hmm... ImageOpen() will use a mix of `sg_DiskImageHelper` and `sg_HardDiskImageHelper` calls (when `bIsHarddisk` is true). Not good!

Also the if-statement tidied up above could be removed and replaced with a polymorphic call to `(CImageHelperBase*)->Open()`.

**Comment by tomcw** [comment]
Refactored & fixed in 3edb782.
